colcon_core/environment/path.py

```diff
@@ -45,3 +46,41 @@ def _create_environment_hooks(
                 break
 
         return hooks
+
+
+class PythonScriptsPathEnvironment(EnvironmentExtensionPoint):
+    """Extend the `PATH` variable to find python scripts."""
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(
+            EnvironmentExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+
+    def create_environment_hooks(self, prefix_path, pkg_name):  # noqa: D102
+        hooks = self._create_environment_hooks(prefix_path, pkg_name, 'bin')
+        if sys.platform == 'win32':
+            hooks += self._create_environment_hooks(
+                prefix_path, pkg_name, 'Scripts', '-scripts')
+        return hooks
+
+    def _create_environment_hooks(
+        self, prefix_path, pkg_name, subdirectory, suffix=''
+    ):
+        hooks = []
+        bin_path = get_python_install_path('scripts', {'base': prefix_path})
+        logger.log(1, "checking '%s'" % bin_path)
+        try:
+            names = os.listdir(str(bin_path))
+        except FileNotFoundError:
+            pass
+        else:
+            for name in names:
+                if not (bin_path / name).is_file():
+                    continue
+                rel_bin_path = bin_path.relative_to(prefix_path)
+                hooks += shell.create_environment_hook(
+                    'path' + suffix, prefix_path, pkg_name, 'PATH',
```

Comment:
I believe that the first argument to `create_environment_hook` is an ID that is expected to be unique, so I believe it should be different from what's passed in the `PathEnvironment` extension.

Reply:
I used `pythonscriptspath` as the id in one of the recent commits